core/simulations/index.ts

```diff
@@ -0,0 +1,46 @@
+import prompts, { PromptObject } from 'prompts';
+import { SIMULATIONS } from './config';
+import readline from 'readline';
+import { HARDHAT_PRIVATE_KEY, HARDHAT_PUBLIC_KEY } from '../helpers/constants';
+
+const keypress = async () => {
+    await prompts({
+        type: 'invisible',
+        name: 'irrelevant',
+        message: 'Press Enter to continue',
+    });
+};
+
+const selectAndRunSimulation = async () => {
+    readline.emitKeypressEvents(process.stdin);
+    const promtChoices = SIMULATIONS.map(simulationConfig => simulationConfig.title).map(simulationName => ({
+        title: simulationName,
+        value: simulationName,
+    }));
+    const promtConfig: PromptObject = {
+        type: 'select',
+        name: 'value',
+        message: 'Select Simulation',
+        choices: promtChoices,
+    };
+    console.info(`Wallet Credentials: public key: ${HARDHAT_PUBLIC_KEY}, private key: ${HARDHAT_PRIVATE_KEY}`);
```

Comment:
I think this can be printed inside the function that resets the network and creates provider? It's kind of out of context here (and if you have provider, you don't have to hardcode public key)